iOS/Flows/Home/Members/OrbCollectionViewLayout.swift

```diff
@@ -28,31 +29,68 @@ class OrbCollectionViewLayout: UICollectionViewLayout {
         return self.collectionView?.numberOfItems(inSection: 0) ?? 0
     }
 
-    var a: CGFloat {
-        return 2.5 * UIScreen.main.bounds.width
-    }
-
-    var b: CGFloat {
-        return 2.5 * UIScreen.main.bounds.height
-    }
-
-    let c: CGFloat = 15
-
     override func shouldInvalidateLayout(forBoundsChange newBounds: CGRect) -> Bool {
         return true
     }
 
+    /// A cache of all the possible positions for the items. Indices correspond to item IndexPaths.
+    private var itemPositions: [CGPoint] = []
+
     override var collectionViewContentSize: CGSize {
         guard let collectionView = self.collectionView else { return .zero }
+        return CGSize(width: collectionView.width.doubled + 500,
+                      height: collectionView.height.doubled + 500)
+    }
+
+    override func prepare() {
+        super.prepare()
+
+        self.itemPositions.removeAll()
 
-        let radius = self.getRadius(forIndexPath: IndexPath(item: self.cellCount - 1, section: 0))
-        let width: CGFloat = radius.doubled + collectionView.halfWidth
-        let height: CGFloat = radius.doubled + collectionView.halfHeight
-        return CGSize(width: width, height: height)
+        let centerPoint = CGPoint(x: self.collectionViewContentSize.width.half,
+                                  y: self.collectionViewContentSize.height.half)
+
+        var ringIndex: Int = 0
+
+        // Generate points that describe the rings of this layout.
+        while self.itemPositions.count < self.cellCount {
```

Comment:
This is logic is similar to before except that instead of laying things out along a circular path, it's a hexagon.